repository/src/main/java/org/apache/atlas/repository/graph/GraphHelper.java

```diff
@@ -817,6 +817,24 @@ public static List<AtlasVertex> getPropagatableClassifications(AtlasEdge edge) {
 
         return ret;
     }
+    //Returns the vertex from which the tag is being propagated
+    public static AtlasVertex getPropagatingVertex(AtlasEdge edge) {
+        if(edge != null) {
+            PropagateTags propagateTags = getPropagateTags(edge);
+            AtlasVertex   outVertex     = edge.getOutVertex();
+            AtlasVertex   inVertex      = edge.getInVertex();
+
+            if (propagateTags == PropagateTags.ONE_TO_TWO || propagateTags == PropagateTags.BOTH) {
+                return outVertex;
+            }
+
+            if (propagateTags == PropagateTags.TWO_TO_ONE || propagateTags == PropagateTags.BOTH) {
+                return inVertex;
+            }
+
+        }
+        return null;
```

Comment:
@sumandas0  **Bad Practice:** Why returning a `null` ? Cant we throw exception and handle ?

Reply:
ok

Reply:
As returning null here signifies that no propagating vertex is there, it might not be bad design. If we don't return this we will not be able to understand if vertex exists there or not.
I also added check when we are using that function, if it return null then we return.